### PR TITLE
{Mixed-Reality} Migration to CodeGen V2

### PR DESCRIPTION
**Pull request by calvinhzy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/mixed-reality/azext_mixed_reality/custom.py [hunk 1]
```diff
@@ -7,171 +7,77 @@
...
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg, AAZArgEnum
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.key = AAZStrArg(
+            options=["--key", "-k"],
+            help="Key to be regenerated.",
+            default="primary"
+        )
+        args_schema.key.enum = AAZArgEnum({"primary":"primary", "secondary":"secondary"})
```
Comment (kairu-ms) [inline src/mixed-reality/azext_mixed_reality/custom.py:54]:
```suggestion
        args_schema.key = AAZStrArg(
            options=["--key", "-k"],
            help="Key to be regenerated.",
            default="primary",
            enum = AAZArgEnum({"primary":"primary", "secondary":"secondary"})
        )
```

File: src/mixed-reality/azext_mixed_reality/custom.py [hunk 2]
```diff
@@ -7,171 +7,77 @@
...
```
Comment (kairu-ms) [inline src/mixed-reality/azext_mixed_reality/custom.py:75]:
```suggestion
        args_schema.key = AAZStrArg(
            options=["--key", "-k"],
            help="Key to be regenerated.",
            default="primary",
            enum=AAZArgEnum({"primary":"primary", "secondary":"secondary"})
        )
```

File: src/mixed-reality/azext_mixed_reality/custom.py [hunk 3]
```diff
@@ -7,171 +7,77 @@
...
+        args_schema.serial._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.key):
+            args.serial = 1 if str(args.key).lower() == 'primary' else 2
+            del args.key
```
Comment (kairu-ms) [inline src/mixed-reality/azext_mixed_reality/custom.py:83]: You don't always need to delete it if it's an additional arg

File: src/mixed-reality/azext_mixed_reality/custom.py [hunk 4]
```diff
@@ -7,171 +7,77 @@
...
+
+logger = get_logger(__name__)
+
+
+class SpatialAnchorsCreate(_SpatialAnchorsCreate):
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.kind):
+            args.sku = args.kind
+            del args.kind
```
Comment (kairu-ms) [inline src/mixed-reality/azext_mixed_reality/custom.py:26]: Why need to delete the kind?

File: src/mixed-reality/azext_mixed_reality/custom.py [hunk 5]
```diff
@@ -7,171 +7,77 @@
...
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.identity._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.kind):
+            args.sku = args.kind
+            del args.kind
```
Comment (kairu-ms) [inline src/mixed-reality/azext_mixed_reality/custom.py:40]: Same concern here

File: src/mixed-reality/azext_mixed_reality/custom.py (hunk 4 above)
Comment (calvinhzy) [reply src/mixed-reality/azext_mixed_reality/custom.py:26]: It would raise an exception if sent with request to service.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**